cpp/src/arrow/compute/api_vector.h

```diff
@@ -292,19 +289,20 @@ ARROW_EXPORT
 Result<std::shared_ptr<Array>> NthToIndices(const Array& values, int64_t n,
                                             ExecContext* ctx = NULLPTR);
 
-/// \brief Returns the first k elements ordered by `options.keys`.
+/// \brief Returns the indices of the first k elements ordered by `options.sort_keys`.
 ///
-/// Return a sorted array with its elements rearranged in such
+/// Select an array of indices of the sorted array with its elements rearranged in such
 /// a way that the value of the element in k-th position (options.k) is in the position it
-/// would be in a sorted datum ordered by `options.keys`. Null like values will be not
-/// part of the output. Output is not guaranteed to be stable.
+/// would be in a sorted datum ordered by `options.sort_keys`. Null like values will be
+/// not part of the output. Output is not guaranteed to be stable.
```

Comment:
I think this wording could probably be improved a little more, perhaps something like (in line with the wording for SortIndices below)

```
Returns the indices that would select the first K elements of the array in the specified order.

Perform an indirect sort of the datum, keeping only the first K elements. The output array will
contain indices such that the item indicated by the k-th index will be in the position it would be
if the datum were sorted by `options.sort_keys`. However, indices of null values will not be
part of the output. The sort is not guaranteed to be stable.
```